File: frontend/src/types/shipment.ts
export interface IShipment {
	id: string | null
	date?: string
	product: string | null
	count: number | null
	money: number | null
}

export interface IShipmentDTO {
	id: string
	date: number
	product: string
	count: string
	money: string
}

export interface IShipmentFull {
	id: string
	date: string
	product: string
	count: number
	money: number
	planQuantity: number
	planMoney: number
}

File: backend/internal/models/shipment.go
package models

type Shipment struct {
	Id      string `json:"id" db:"id"`
	Date    int64  `json:"date" db:"date" binding:"required"`
	Product string `json:"product" db:"product" binding:"required"`
	Count   string `json:"count" db:"count"`
	Money   string `json:"money" db:"money"`
}

type ShipmentGrouped struct {
	Date     int64      `json:"date"`
	Shipment []Shipment `json:"shipment"`
}
